api/api/routers/mcp/_mcp_service.py

```diff
@@ -629,6 +631,118 @@ async def deploy_agent_version(
                 error=f"Failed to deploy version: {str(e)}",
             )
 
+    async def search_documentation(
+        self,
+        query: str | None = None,
+        page: str | None = None,
+    ) -> LegacyMCPToolReturn:
+        """Search WorkflowAI documentation OR fetch a specific documentation page.
+
+        Args:
+            query: Search across all documentation to find relevant content snippets
+            page: Direct access to specific documentation page
+
+        Returns:
+            LegacyMCPToolReturn with either search results or page content
+        """
+        if query and page:
+            return LegacyMCPToolReturn(
+                success=False,
+                error="Use either 'query' OR 'page' parameter, not both",
+            )
+
+        if not query and not page:
+            return LegacyMCPToolReturn(
+                success=False,
+                error="Provide either 'query' or 'page' parameter",
+            )
+
+        if query:
+            return await self._search_documentation_by_query(query)
+
+        if page:
+            return await self._get_documentation_page(page)
+
+        # This should never be reached due to the parameter validation above
+        return LegacyMCPToolReturn(
+            success=False,
+            error="Invalid parameters provided",
+        )
+
+    async def _search_documentation_by_query(self, query: str) -> LegacyMCPToolReturn:
+        """Search documentation using query and return snippets."""
+        try:
+            documentation_service = DocumentationService()
+            relevant_sections = await documentation_service.get_relevant_doc_sections(
+                chat_messages=[ChatMessage(role="USER", content=query)],
+                agent_instructions="",
+            )
+
+            # Convert to SearchResult format with content snippets
+            search_results: list[dict[str, str]] = [
+                {
+                    "content_snippet": section.content,
+                    "source_page": section.title.replace("content/", ""),
+                }
+                for section in relevant_sections
+            ]
+
+            if len(search_results) == 0:
+                return LegacyMCPToolReturn(
+                    success=True,
+                    messages=[f"No relevant documentation sections found for query: {query}"],
+                )
+
+            return LegacyMCPToolReturn(
+                success=True,
+                data={"search_results": search_results},
+                messages=[
+                    f"Successfully found relevant documentation sections: {[section.title for section in relevant_sections]}",
+                ],
+            )
+
+        except Exception as e:
```

Comment:
we should have exception catch all. It's ok to just raise an exception here no since it's an internal error